sigle/src/modules/analytics/stats/StatsFrame.tsx

```diff
@@ -0,0 +1,178 @@
+import { eachDayOfInterval, format } from 'date-fns';
+import { useEffect, useState } from 'react';
+import { AnalyticsHistoricalResponse, StatsData } from '../../../types';
+import {
+  Box,
+  Flex,
+  Tabs,
+  TabsContent,
+  TabsList,
+  TabsTrigger,
+  Text,
+} from '../../../ui';
+import { StatsChart } from './StatsChart';
+import { FATHOM_MAX_FROM_DATE } from '../../../pages/api/analytics/utils';
+
+// prevent flash of no content in graph by initializing range data with a constant value (1)
+const today = new Date();
+const weekFromDate = new Date(new Date().setDate(today.getDate() - 7));
+const monthFromDate = new Date(new Date().setDate(today.getDate() - 30));
+
+const dates = eachDayOfInterval({
+  start: weekFromDate,
+  end: today,
+});
+
+const initialRange: StatsData[] = dates.map((date) => {
+  return {
+    value: 0,
+    date: date.toString(),
+    visits: 0,
+  };
+});
+
+export const StatsFrame = () => {
+  const [data, setData] = useState<StatsData[]>(initialRange);
+
+  const baseUrl = window.location.origin;
+
+  useEffect(() => {
+    fetchWeeklyStats();
+  }, []);
+
+  const fetchWeeklyStats = async () => {
```

Comment:
by looking at the code it feels like fetchWeeklyStats, fetchMonthlyStats and fetchAllTimeStats could be merged into one function as they are doing the same thing, wdyt?

Reply:
agreed, will merge them :)